Added clear button and refractored code

ravjot07 opened the pull request:
Added clear button that Clears the canvas and reset the drawing lines. Contribution towards #14 

https://github.com/user-attachments/assets/d7fd9ea0-9934-4dca-8816-cda3de80121b

ravjot07 commented:
@quozl can you review this pr

chimosky commented on js/activity.js:
```diff
@@ -2,32 +2,43 @@ define(function (require) {
         }
-        // Make the activity stop with the stop button.
+
+        // Handle Stop Button Click
```
Comments like this are unnecessary, and not needed.
Please remove them.

ravjot07 replied:
i have removed unnecessary comments

chimosky commented on css/activity.css:
```diff
@@ -5,3 +5,7 @@
+
+#main-toolbar #clear-button {
+  background-image: url(../icons/clear-button.svg);
+}
```
You can remove the No new line at EOF by running `echo >> css/activity.css`, this makes it better to view when using diff tools or cat.

ravjot07 replied:
done...

chimosky commented on js/activity-works.js:
```diff
@@ -1,10 +1,12 @@
 define(function (require) {
     var activity = require("sugar-web/activity/activity");
     var icon = require("sugar-web/graphics/icon");
+    require("easel");
+    require("handlebars");
+    var shapes = require("activity/shapes");
```
Are these used here?

chimosky commented on js/activity-works.js:
```diff
@@ -41,231 +54,247 @@ define(function (require) {
-
-                // enabled mouse over / out events
-                stage.enableMouseOver(10);
+        // Get things started
+        init();
```
Please remove redundant comments like this, they're not useful.

ravjot07 replied:
done..

chimosky commented on js/activity.js:
```diff
@@ -2,32 +2,43 @@ define(function (require) {
         // Then create a list of the label elements
-        nlabels = [];
-        for (i = 0; i < 21; i++) {
+        var nlabels = [];
```
If this is consistent throughout this code, there's no need to change it.

ravjot07 replied:
sure i will keep them as it is....

chimosky commented:
See #15 to make sure you've addressed the concerns listed there, reviewed, not tested.

The code refactoring shouldn't be in the same commit as the clear button change, keep both separate as it's easier to review that way.

ravjot07 commented on js/activity.js:
```diff
@@ -2,32 +2,43 @@ define(function (require) {
-        nlabels = [];
-        for (i = 0; i < 21; i++) {
+        var nlabels = [];
+        for (var i = 0; i < 21; i++) {
```
done...

ravjot07 commented:
> See #15 to make sure you've addressed the concerns listed there, reviewed, not tested.
> 
> The code refactoring shouldn't be in the same commit as the clear button change, keep both separate as it's easier to review that way.

yes  i have addressed the concerns listed in #15.... mainly they discussed about position of the button and its icon visibility and its functioning  about deleting only the drawing rather than whole canvas...... i have covered all of that

ravjot07 commented:
@chimosky are there any more changes or suggestion or we should merge this PR

chimosky commented:
Would be great to have better commit messages than "Minor Updates", see [making commits](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits).

ravjot07 commented:
> Would be great to have better commit messages than "Minor Updates", see [making commits](https://github.com/sugarlabs/sugar-docs/blob/master/src/contributing.md#making-commits).
> 
> I'll test this once I can.

@chimosky sure i will take care of commit msg next time.

quozl commented:
You can and should change commit messages now rather than wait for next time.

ravjot07 commented:
> You can and should change commit messages now rather than wait for next time.

Done @quozl

quozl commented:
Tested.  Reviewed.
